### PR TITLE
Feature/#112 slot annotator interface

### PR DESCRIPTION
**Pull request by NoB0**
Create an interface for slot annotators. Upon approval,`NLU` should be refactored to take `List[SlotAnnotator]` for `slot_annotators` argument.

Part of #112

### CODE REVIEW

File: dialoguekit/nlu/nlu.py [hunk 1]
```diff
@@ -10,12 +10,16 @@
 
 class NLU:
     def __init__(
-        self, intent_classifier: IntentClassifier, slot_annotators: List[Any]
+        self,
+        intent_classifier: IntentClassifier,
+        slot_annotators: List[Any],
```
Comment (IKostric) [inline dialoguekit/nlu/nlu.py:15]:
```suggestion
        slot_annotators: List[SlotAnnotator],
```

File: dialoguekit/nlu/nlu.py [hunk 2]
```diff
@@ -26,7 +30,7 @@ def classify_intent(self, utterance: Utterance) -> Intent:
     def annotate_slot_values(
         self, utterance: Utterance
     ) -> List[SlotValueAnnotation]:
-        """Annotates a given utterance."""
+        """Annotates a given utterance with slot annotators."""
```
Comment (IKostric) [inline dialoguekit/nlu/nlu.py:33]: Missing `Args:` and `Returns:`.

**Review by kbalog — APPROVED** [review]
LGTM (but agree with Ivica's comments that should be addressed)

**Review by IKostric — APPROVED** [review]
LGTM,
Make sure all tests pass before merging.

**Pull request merged**